### PR TITLE
actually expose the dns port in the skydns container...

### PR DESCRIPTION
**Pull request by AntonioMeireles**
without this (and unless i'm missing something obvious) _thing_ not as _usefull_ as it could ...

[/cc @pires whom without his unvaluable help i'd still be chasing gooses on this]
/cc @thockin

### CODE REVIEW

**Comment by pires** [comment]
Hope this fixes #7389

**Comment by a-robinson** [comment]
This looks like it's for the best, but out of curiosity, how did you start your skydns container? The default [skydns replication controller](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/addons/dns/skydns-rc.yaml.in#L50) (and its corresponding service) does listen on port 53.

**Comment by pires** [comment]
@a-robinson can you please verify that it really works? Cause I'm using the exact same file and it's not working.

**Comment by AntonioMeireles** [comment]
@a-robinson - `docker inspect` of the skydns container returns no open ports. {w,s}ould whatever ports one place in the controller matter if the actuall container hasn't them exposed ?

**Comment by pires** [comment]
Here's `skydns` logs
```
$ kubectl log k8s-dns-rtde1 skydns
2015-04-27T17:26:56.370866670Z 2015/04/27 17:26:56 skydns: falling back to default configuration, could not read from etcd: 100: Key not found (/skydns/config) [5]
2015-04-27T17:26:56.370866670Z 2015/04/27 17:26:56 skydns: ready for queries on k8s.local. for tcp://0.0.0.0:53 [rcache 0]
2015-04-27T17:26:56.370866670Z 2015/04/27 17:26:56 skydns: ready for queries on k8s.local. for udp://0.0.0.0:53 [rcache 0]
```

Yet, I enter the container and
```
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                                            COMMAND                CREATED             STATUS              PORTS               NAMES
d576e19ad5e5        gcr.io/google_containers/skydns:2015-03-11-001   "/skydns -machines=h   4 hours ago         Up 4 hours                              k8s_skydns.fb57278f_k8s-dns-rtde1_default_66644233-ed02-11e4-bc34-0800279924c6_8e5fe105
02709b80ed72        gcr.io/google_containers/kube2sky:1.2            "/kube2sky -etcd-ser   4 hours ago         Up 4 hours                              k8s_kube2sky.84dc01ab_k8s-dns-rtde1_default_66644233-ed02-11e4-bc34-0800279924c6_3ce05ff3
6710ce9c0576        quay.io/coreos/etcd:v2.0.3                       "/etcd -listen-clien   4 hours ago         Up 4 hours                              k8s_etcd.97d6179f_k8s-dns-rtde1_default_66644233-ed02-11e4-bc34-0800279924c6_bf86e0e8
097df6058d05        gcr.io/google_containers/pause:0.8.0             "/pause"               4 hours ago         Up 4 hours                              k8s_POD.48fc0515_k8s-dns-rtde1_default_66644233-ed02-11e4-bc34-0800279924c6_5d958925
core@node-01 ~ $ docker exec -it d57 sh
/ # netstat ae
Active Internet connections (w/o servers)
Proto Recv-Q Send-Q Local Address           Foreign Address         State
tcp        0      0 k8s-dns-rtde1:54318     10.100.0.1:www          ESTABLISHED
tcp        0      0 localhost:55630         localhost:4001          ESTABLISHED
tcp        0      0 localhost:55601         localhost:4001          ESTABLISHED
tcp        0      0 localhost:55602         localhost:4001          ESTABLISHED
Active UNIX domain sockets (w/o servers)
Proto RefCnt Flags       Type       State         I-Node Path
```

No 53 ports anywhere.

**Comment by AntonioMeireles** [comment]
fwiw 
```
core@node-01 ~ $ docker ps
CONTAINER ID        IMAGE                                            COMMAND                CREATED             STATUS              PORTS               NAMES
851c28c7e330        gcr.io/google_containers/skydns:2015-03-11-001   "/skydns -machines=h   16 seconds ago      Up 15 seconds                           k8s_skydns.e2143314_k8s-dns-0lnko_default_92878d67-ed1e-11e4-b4c1-001c42100c0f_07b4df54   
f0bb7e52ca87        quay.io/coreos/etcd:v2.0.7                       "/etcd -listen-clien   16 seconds ago      Up 15 seconds                           k8s_etcd.a30217a3_k8s-dns-0lnko_default_92878d67-ed1e-11e4-b4c1-001c42100c0f_b3d00f86     
dd0ec29e2960        gcr.io/google_containers/pause:0.8.0             "/pause"               16 seconds ago      Up 15 seconds                           k8s_POD.48fc0515_k8s-dns-0lnko_default_92878d67-ed1e-11e4-b4c1-001c42100c0f_8a4049c0      
core@node-01 ~ $ docker exec -ti 851c28c7e330 /bin/sh 
/ # nslookup k8s-dns.default.k8s.local localhost
Server:    127.0.0.1
Address 1: 127.0.0.1 localhost

Name:      k8s-dns.default.k8s.local
Address 1: 10.100.0.10
```

File: cluster/addons/dns/skydns/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,5 @@
 FROM busybox
 MAINTAINER Tim Hockin <thockin@google.com>
 ADD skydns skydns
+EXPOSE 53 53/udp
```
Comment (thockin) [inline cluster/addons/dns/skydns/Dockerfile]: This is not required in kubernetes, it is purely advisory.

**Comment by thockin** [comment]
Please explain?  DNS works fine without this for everyone else. It seems that we don't pass the ports down to docker ( @dchen1107 ) but the nature of pod IPs should not require it.  What are your docker daemon flags?

**Comment by scott-abernethy** [comment]
Just had the same issue, and using a custom skydns image with the ports exposed resolved the issue.

My setup:
* AWS cluster
* CoreOS version 633.1.0
* Docker version 1.5.0, build a8a31ef-dirty

Let me know what info you need to debug.

**Comment by thockin** [comment]
ps auxw | grep docker

**Comment by pires** [comment]
```
docker --daemon --host=fd:// --bip=10.244.55.1/24 --mtu=1450 --ip-masq=false
```

**Comment by scott-abernethy** [comment]
docker --daemon --host=fd:// --insecure-registry=172.31.9.67:5000 --registry-mirror=http://172.31.9.67:5001 --bip=10.244.15.1/24 --mtu=8951 --ip-masq=false

**Comment by scott-abernethy** [comment]
I moved my comment to #7389 ... and added that I'm using Kubernetes 0.11.0.

**Comment by scott-abernethy** [comment]
Full output
$ ps auxw | grep docker
root       582  0.0  0.5 684460 20980 ?        Ssl  Apr27   0:01 docker --daemon --host=fd:// --bridge=none --iptables=false --ip-masq=false --graph=/var/lib/early-docker --pidfile=/var/run/early-docker.pid
root       616  0.0  0.0 176252  3232 ?        Ssl  Apr27   0:00 /usr/libexec/sdnotify-proxy /run/flannel/sd.sock /usr/bin/docker run --net=host --privileged=true --rm --volume=/run/flannel:/run/flannel --env=NOTIFY_SOCKET=/run/flannel/sd.sock --env-file=/run/flannel/options.env --volume=/etc/ssl/etcd:/etc/ssl/etcd:ro quay.io/coreos/flannel:0.3.0 /opt/bin/flanneld --ip-masq=true
root       622  0.0  0.3 204776 13148 ?        Sl   Apr27   0:00 /usr/bin/docker run --net=host --privileged=true --rm --volume=/run/flannel:/run/flannel --env=NOTIFY_SOCKET=/run/flannel/sd.sock --env-file=/run/flannel/options.env --volume=/etc/ssl/etcd:/etc/ssl/etcd:ro quay.io/coreos/flannel:0.3.0 /opt/bin/flanneld --ip-masq=true
root       685  0.8  1.3 1313168 54016 ?       Ssl  Apr27   1:22 docker --daemon --host=fd:// --insecure-registry=172.31.9.67:5000 --registry-mirror=http://172.31.9.67:5001 --bip=10.244.15.1/24 --mtu=8951 --ip-masq=false

**Comment by thockin** [comment]
Fascinating.  Did it ever work for you folks?  It works just fine for me.

**Comment by scott-abernethy** [comment]
We only tried to turn on the cluster DNS add-on today, and couldn't get it working until this thread was found and we added the `EXPOSE 53 53/udp` to your skydns docker. I'll let you know what happens after we upgrade to 0.15.0.

**Comment by AntonioMeireles** [comment]
this is not a bug at all. things are (and were) working as advertised. just subtle typos in my and @pires dns machinery. the fact that netstat and lsof weren't that usefull only added fog and fuzzyness to the thing. closing.

@thockin sorry for the *goose chase*, my bad :cry:

**Comment by AntonioMeireles** [comment]
closing per previous comment

**Pull request closed without merging**